src/codegen/gencommon/closuresToClass.ml

```diff
@@ -278,8 +278,9 @@ let traverse gen ?tparam_anon_decl ?tparam_anon_acc (handle_anon_func:texpr->tfu
 
 let rec get_type_params acc t =
 	match t with
-		| TInst(( { cl_kind = KTypeParameter _ } as cl), []) ->
-			if List.memq cl acc then acc else cl :: acc
+		| TInst(( { cl_kind = KTypeParameter constraints } as cl), []) ->
+			let params = List.fold_left get_type_params acc constraints in
+			List.filter (fun t -> not (List.memq t acc)) (cl :: params) @ acc;
```

Comment:
I don't really understand what's going on here and what's the purpose of this function. I see it basically collects all type params, but what this change actually does? Perhaps if you have a good understanding maybe you could comment it a bit for future generations? :)))

Reply:
It may well be nonsense and only work by chance in my tests, I'm really not familiar with ocaml yet.

It collects all type params and runs again `get_type_params` on their constraints to be sure to have all necessary types. In the second line, it adds only types that are not already in `acc`, as before (except that now it runs that on both `cl` and `params` instead of just `cl`).

Reply:
Yeah I understood that far :) I'm just wondering how is this used after. But oh well, I guess I'll have to spend an evening with a camomile tea to get an idea of what's going on with type params in gencommon in general :)